.github/workflows/cygwin.yml

```diff
@@ -49,7 +50,7 @@ jobs:
         id: restore-cache
         with:
           # should use 'pip3 cache dir' to discover this path
-          path: C:\cygwin\home\runneradmin\.cache\pip
+          path: ${{ matrix.INSTALL_DIR }}\home\runneradmin\.cache\pip
```

Comment:
Looking at the comment above this line, this was intended to connect to the pip existing cache location as indicated by `pip3 cache dir` command.

Also using `${{ github.workspace }}` should always be preferable, instead of hard-coding D:\ drive which inevitably will break one day.

Reply:
are you suggesting I move the cygwin install action in front of the cache restore, run pip cach dir, store it as an output and re-use it here?

Reply:
I would suggest using `PIP_CACHE_DIR` and set it to `${{ github.workspace }}/.cache/pip`. Or use `pip --cache-dir` because I think cygwin shell doesn't inherit environment .

As for cygwin installation path, I would think it should be fixed upstream. But here I would suggest using `install-dir: ${{ github.workspace }}/cygwin`

Reply:
> cygwin install action in front of the cache restore, run pip cache dir, store it as an output and re-use it here?

Yeah, that's probably what it should do, but the voodoo to capture the output from a step escaped me at the time I wrote it...

Reply:
> As for cygwin installation path, I would think it should be fixed upstream

Someone is working on this change https://github.com/cygwin/cygwin-install-action/pull/27